openmicroscopy/openmicroscopy
# Reverse invert webclient

# What this PR does

Renames "reverse" to "inverted" in the url ```map``` query string parameter.
```
?maps=[{"inverted":{"enabled":false}},{"inverted":{"enabled":true}}]
```
Also adds "inverted" to json for the webgateway/imgData/ json, but kept the "reverseIntensity" boolean too, so that ```iviewer``` etc aren't broken. 
```
{ "reverseIntensity": true
  "inverted": true,
```
same for ```conn. getAllRenderingDefs()``` ( -> webgateway/get_image_rdefs_json/<id>).


# Testing this PR
This affects all rendering settings functionality in webclient.

1. Choose "Invert" channel
1. Test copy, paste of settings between images
1. Pick rendering settings from owner's thumbnails
1. Undo, redo etc.
1. Open Full viewer from Preview panel - url should have ```?maps=[{"inverted":{"enabled":false}},{"inverted":{"enabled":true}}]``` and image should respect same invert settings as in Preview panel.
1. Copy image link in Full viewer. Should have "inverted" in url. Pasting into new window,"inverted" should be respected.
1. iviewer should behave as normal
1. /webgateway/imgData/<id> should show "inverted" flag in channels as well as "reverseIntensity" (duplicated)
